qutebrowser/browser/greasemonkey.py

```diff
@@ -356,40 +399,73 @@ def _add_script_with_requires(self, script, quiet=False):
         """
         # See if we are still waiting on any required scripts for this one
         for dl in self._in_progress_dls:
-            if dl.requested_url in script.requires:
+            if dl.requested_url in script.requires or \
+                    dl.requested_url in script.resources:
```

Comment:
`script.resources` holds _Resource objects, not strings or URLs so this'll never pick up pending resource downloads. I was reliably getting crashes due to some assertion error on duplicates when force reloading the "Wanikani Niai Similar Kanji" script.

Reply:
(unfortunately this function is not type-checked...)